src/workspace.ts

```diff
@@ -8,26 +8,21 @@ import IModel = monaco.editor.IModel;
 
 export class MonacoWorkspace implements Workspace {
 
-    protected _rootUri: string | null = null;
-
     protected readonly documents = new Map<string, TextDocument>();
     protected readonly onDidOpenTextDocumentEmitter = new Emitter<TextDocument>();
     protected readonly onDidCloseTextDocumentEmitter = new Emitter<TextDocument>();
     protected readonly onDidChangeTextDocumentEmitter = new Emitter<TextDocumentDidChangeEvent>();
 
     constructor(
-        protected readonly m2p: MonacoToProtocolConverter) {
+        protected readonly m2p: MonacoToProtocolConverter,
+        public readonly rootUri: string | null = null) {
```

Comment:
It is a breaking change there are subclasses which reassign `_rootUri`. It would be better to leave `rootUri` getter and just move `protected _rootUri: string | null = null` to the constructor.